manifest: samples: boards: nrf: ieee802154: Optimize power on nRF53

pszkotak opened the pull request:
Optimize power usage of the net core in both single
and multiprotocol configurations by disabling unused
peripherals on the nRF53 platform.

KRKNWK-8952

https://github.com/nrfconnect/sdk-zephyr/pull/463

Signed-off-by: Piotr Szkotak <piotr.szkotak@nordicsemi.no>

NordicBuilder commented:
You can find the documentation preview for this PR at this <a href='http://developer.nordicsemi.com/nRF_Connect_SDK_dev/doc/PR-3940
'>link</a>. It will be updated about 10 minutes after the documentation build succeeds.

*Note: This comment is automatically posted by the Documentation Publishing GitHub Action.*

lmaciejonczyk commented:
I see that we disable UART for net core only for coap samples (all variants). What about the other samples? What is the approach we have here? Disabling UART for child image means no logging from net core. I'm missing this information in the documentation here.

lmaciejonczyk reviewed (commented):
I have two questions:
1. I see that we disable UART for net core only for coap samples (all variants). What about the other samples? What is the approach we have here?
2. Disabling UART for child image means no logging from net core. I'm missing this information in the documentation here.

LuDuda commented on samples/openthread/coap_client/boards/nrf5340dk_nrf5340_cpuapp.overlay:
```diff
@@ -0,0 +1,53 @@
+ */
+
+&adc {
+	status = "disabled";
```
Why those peripherals are enabled in the first place? It seems super strange to disable tens of peripherals for each example.

pszkotak replied:
These are enabled by default, aren't they?
It is a [common practice](https://github.com/nrfconnect/sdk-nrf/blob/master/samples/openthread/coap_client/boards/nrf52840dk_nrf52840.overlay) to disable unused peripherals for coap_client sample.

lmaciejonczyk replied:
Everything is enabled by default to limit actions needed to activate a chosen feature. The case for coap_client sample, especially MTD variant, is to present the minimum current consamption. These is why we are disabling unused peripheral by this overlay for this sample.

LuDuda replied:
I just questioned scheme of enabling peripherals by default (and if I'm not mistaken, only selected peripherals) - not the necessity of this PR.

Is this really the case in Zephyr for all MCUs? I would expect applications to enable peripherals only if they are needed, not in another way around.

pszkotak commented:
> I have two questions:
> 
> 1. I see that we disable UART for net core only for coap samples (all variants). What about the other samples? What is the approach we have here?
> 2. Disabling UART for child image means no logging from net core. I'm missing this information in the documentation here.

Thanks Łukasz,
1. Fixed that by removing child image overlays and providing a single overlay for the net core firmware.
2. Added information to the readme.

lmaciejonczyk reviewed (approved):
LGTM

ru-fu commented on samples/nrf5340/multiprotocol_rpmsg/README.rst:
```diff
@@ -19,6 +19,10 @@ See the following configuration options for more information:
 You might need to adjust the Kconfig configuration of this sample to make it compatible with the peer application.
 For example, :option:`CONFIG_BT_MAX_CONN` must be equal to the maximum number of connections supported by the peer application.
 
+The sample is configured to be energy efficient by disabling:
+ * Serial Console and Logger in :file:`prj.conf`
+ * Unused peripherals in :file:`boards/nrf5340dk_nrf5340_cpunet.overlay`
```
```suggestion
The following components have been disabled to make this sample energy-efficient:

* Serial console (in :file:`prj.conf`)
* Logger (in :file:`prj.conf`)
* Unused peripherals (in :file:`boards/nrf5340dk_nrf5340_cpunet.overlay`)
```

pszkotak replied:
The case is that Logger is not explicitly disabled in the `prj.conf`, it is just not enabled by anything.
I would like to avoid a situation in which a customer would look in the `prj.conf` searching for specific line disabling logging.

ru-fu replied:
I thought this is what CONFIG_LOG=n in there does?

pszkotak replied:
It's not there https://github.com/nrfconnect/sdk-zephyr/blob/2ccd9a9a5218a4c10ff61528d585e31f4f2cbeb7/samples/boards/nrf/ieee802154/802154_rpmsg/prj.conf

pszkotak replied:
Done. The last line was removed as the `nrf5340dk_nrf5340_cpunet.overlay` is no longer included in this PR.

ru-fu commented on samples/nrf5340/multiprotocol_rpmsg/README.rst:
```diff
@@ -19,6 +19,10 @@ See the following configuration options for more information:
 For example, :option:`CONFIG_BT_MAX_CONN` must be equal to the maximum number of connections supported by the peer application.
 
+The following components have been disabled to make this sample energy-efficient:
+* Serial console (in :file:`prj.conf`)
```
You need an empty line before the list to make it display correctly.

pszkotak replied:
Thanks, I always forget about it...

tejlmand commented:
please rebase and update SHA: d6d0ffde53a88d6de834e07dceac47af8f7687a3
